test/extensions/http/header_validators/envoy_default/http2_header_validator_test.cc

```diff
@@ -181,6 +189,43 @@ TEST_F(Http2HeaderValidatorTest, ValidateRequestHeaderMapDropUnderscoreHeaders)
           {{":scheme", "https"}, {":method", "GET"}, {":path", "/"}, {":authority", "envoy.com"}}));
 }
 
+TEST_F(Http2HeaderValidatorTest, RequestExtendedConnect) {
+  ::Envoy::Http::TestRequestHeaderMapImpl headers{
+      {":scheme", "https"},  {":method", "CONNECT"},      {":protocol", "websocket"},
+      {":path", "/foo/bar"}, {":authority", "envoy.com"}, {"x-foo", "bar"}};
+  auto uhv = createH2(empty_config);
+  EXPECT_ACCEPT(uhv->validateRequestHeaderMap(headers));
+}
+
+TEST_F(Http2HeaderValidatorTest, RequestExtendedConnectNoScheme) {
+  ::Envoy::Http::TestRequestHeaderMapImpl headers{{":method", "CONNECT"},
+                                                  {":protocol", "websocket"},
+                                                  {":path", "/foo/bar"},
+                                                  {":authority", "envoy.com"}};
+  auto uhv = createH2(empty_config);
+  EXPECT_REJECT_WITH_DETAILS(uhv->validateRequestHeaderMap(headers),
+                             UhvResponseCodeDetail::get().InvalidScheme);
+}
+
+TEST_F(Http2HeaderValidatorTest, RequestExtendedConnectNoPath) {
+  ::Envoy::Http::TestRequestHeaderMapImpl headers{{":scheme", "https"},
```

Comment:
also one for extended connect invalid path?

Reply:
Done